### ISSUE TITLE
iOS 14 and Safari 14 using jitsi meet in Safari 14 on iPhone or iPad

### ISSUE THREAD

**Issue by gh264367** [jitsi/jitsi-meet#7953]
I am getting a black box with using jitsi meet on/in iOS14/safari14 with an iPhone or iPad. iPhone or iPad produces a black box to the other person/people in the video session. This is happens on the web version with own jitsi server and happens on the meet.jit.si web app. Is this being looked into?

**Comment by jallamsetty1** [comment]
I can't reproduce this on meet.jit.si using an iPhone running iOS 14.0.1. Can you provide more details about the devices that you are able to reproduce this on ? Was it working before on these devices before upgrading to iOS 14 ?

**Comment by gh264367** [comment]
> 
> 
> I can't reproduce this on meet.jit.si using an iPhone running iOS 14.0.1. Can you provide more details about the devices that you are able to reproduce this on ? Was it working before on these devices before upgrading to iOS 14 ?

Hello jallamsetty1, it was working before on iOS 13.5.1 on an iPhone Xr and iPad 5, both through a mobile service(verizon) and on wifi, using Safari 13.5.1. Since I have upgraded to iOS 14.0.1, I get black boxes through my camera on my iPad/iPhone with each of the device connected via meet.jit.si or my own server. If someone on another device, such as an android device using Chrome or, PC/Mac using Chrome, are connected to either my iPhone or iPad with iOS 14.0.1 using Safari, still get the black boxes from my iphone and ipad. Does this make sense?

**Comment by jallamsetty1** [comment]
> > I can't reproduce this on meet.jit.si using an iPhone running iOS 14.0.1. Can you provide more details about the devices that you are able to reproduce this on ? Was it working before on these devices before upgrading to iOS 14 ?
> 
> Hello jallamsetty1, it was working before on iOS 13.5.1 on an iPhone Xr and iPad 5, both through a mobile service(verizon) and on wifi, using Safari 13.5.1. Since I have upgraded to iOS 14.0.1, I get black boxes through my camera on my iPad/iPhone with each of the device connected via meet.jit.si or my own server. If someone on another device, such as an android device using Chrome or, PC/Mac using Chrome, are connected to either my iPhone or iPad with iOS 14.0.1 using Safari, still get the black boxes from my iphone and ipad. Does this make sense?

Thanks for the info, will try to get hold of an iPhone Xr or iPad 5 to reproduce this issue.

**Comment by gh264367** [comment]
Hey jallamsetty1, thanks for getting back. I am discovering that this is happening in Safari 14 on mobile devices, while using the meet.jit.si web application or on own jitsi meet server. I have tried this with other people on iPhones/iPads with iOS 14 and using Safari 14 for jitsi meet or  meet.jit.si web application. I get black boxes for the video and do hear them from the person using Safari 14 and iOS14 to my Android phone/tablet(Chrome) or PC using Chome.  The people on the Safari 14 are able to see me in the video and hear me. Hope this makes sense. Thanks.

**Comment by nehatdedinca** [comment]
iOS version: 14.1
Browser: Safari

**Comment by gh264367** [comment]
Yes. iOS version 14 and Safari version 14.

**Comment by Jurik81** [comment]
Hello, do we have any updates on this one? Just asking, because it has no assignees yet.

**Comment by saghul** [comment]
We don't assign issues to specific people around here. @jallamsetty1 is the more knowledgeable person in this area, but everyone is busy, so it may take a bit to it to be looked into.

**Comment by gh264367** [comment]
Since this problem is known by Jitsi, is there an estimate time of when a fix for this will be available?

**Comment by nadelicious** [comment]
@jallamsetty1  there's also an issue in IOS safari when you remove the earphones while in a call. As a result, the user can't receive audio anymore from another participant. 

IOS versions: 14.1, 13.7
Devices: iPhone 8, iPhone X
1 on 1 call

**Comment by jallamsetty1** [comment]
> @jallamsetty1 there's also an issue in IOS safari when you remove the earphones while in a call. Afterwards, the user can't receive audio anymore from another participant.
> 
> IOS versions: 14.1, 13.7
> Devices: iPhone 8, iPhone X
> 1 on 1 call

hi @nadelicious, we do not have any control over the audio output device on Safari as it blocked by the browser because of privacy reasons. I will try to check if there is a open Safari bug and update it or open a new one if I am able to reproduce it.

**Comment by neehalsaal** [comment]
Hi @jallamsetty1,
We see that web issue https://github.com/jitsi/jitsi-meet/issues/7984 was marked as a duplicate of this one here. Now that this particular issue is closed in https://github.com/jitsi/lib-jitsi-meet/pull/1400 (lib-jitsi-meet), we expected that the web issue too to have fixed.
But it doesnt seem to have fixed the web issue(on our jitsi deployment, not meet.jit.si) even after taking the latest lib-jitsi-meet(as of 15th November, 2020). Is there any other change that has to be picked up along with the lib-jitsi-meet change? May be some jitsi-meet reactjs change too, that you would like to mention?

Thanks in advance.

**Comment by jallamsetty1** [comment]
meet.jit.si hasn't been updated with this fix yet but it should be available in jitsi-meet-web-4544-1.

**Comment by Yolakalemowa** [comment]
still no support for Safari? :( why?

KEep getting the same message. only supported for chrome and firefox

**Comment by saghul** [comment]
What version are you testing? Safari has been supported for a long time now.

**Comment by Yolakalemowa** [comment]
It's my friend's macbook. It says version 13.1.1 (15609.2.1.2)

isn't that the latest? sorry noooooob here :(

**Comment by saghul** [comment]
That's a too old version of Safari. It's on version 15 now. Due to how fast WebRTC still moves and how aggressive Apple is deprecating old operating system versions we simply can't afford to support such old Safari versions. Our support policy for Safari is "latest minus one": https://jitsi.github.io/handbook/docs/user-guide/supported-browsers

**Comment by Yolakalemowa** [comment]
Understood... and fair enough I guess. 

thanks a lot!
